utils/public_api.py

```
@@ -0,0 +1,824 @@
+
+
+def _emit_output(text: str) -> None:
+    """Emit output directly or through a pager depending on terminal size."""
+    if not _should_page(text):
+        sys.stdout.write(text)
+        return
+
+    pager = shlex.split(os.environ.get('PAGER', 'less -R'))
+
+    try:
+        subprocess.run(pager, input=text, text=True, check=False)
```

Ruff (0.15.21)

[error] `subprocess` call: check for execution of untrusted input

(S603)